crypto/src/main/scala/org/bitcoins/crypto/ECKey.scala

```diff
@@ -122,83 +244,66 @@ sealed abstract class ECPrivateKey
 
   def fieldElement: FieldElement = FieldElement(bytes)
 
-  override def toStringSensitive: String = s"ECPrivateKey($hex,$isCompressed)"
-}
-
-object ECPrivateKey extends Factory[ECPrivateKey] {
-
-  private case class ECPrivateKeyImpl(
-      override val bytes: ByteVector,
-      isCompressed: Boolean,
-      ec: ExecutionContext)
-      extends ECPrivateKey {
-    require(CryptoUtil.secKeyVerify(bytes), s"Invalid key, hex: ${bytes.toHex}")
-  }
+  override def toStringSensitive: String = s"ECPrivateKey($hex)"
 
-  def apply(bytes: ByteVector, isCompressed: Boolean)(implicit
-      ec: ExecutionContext): ECPrivateKey = {
-    ECPrivateKeyImpl(bytes, isCompressed, ec)
+  def toPrivateKeyBytes(isCompressed: Boolean = true): ECPrivateKeyBytes = {
+    ECPrivateKeyBytes(bytes, isCompressed)
   }
+}
 
-  override def fromBytes(bytes: ByteVector): ECPrivateKey =
-    fromBytes(bytes, isCompressed = true)
+object ECPrivateKey extends Factory[ECPrivateKey] {
 
   @tailrec
-  def fromBytes(bytes: ByteVector, isCompressed: Boolean): ECPrivateKey = {
-
+  override def fromBytes(bytes: ByteVector): ECPrivateKey = {
     if (bytes.size == 32)
-      ECPrivateKeyImpl(bytes, isCompressed, ExecutionContext.global)
+      new ECPrivateKey(bytes)
     else if (bytes.size < 32) {
       //means we need to pad the private key with 0 bytes so we have 32 bytes
-      ECPrivateKey.fromBytes(bytes.padLeft(32), isCompressed)
+      ECPrivateKey.fromBytes(bytes.padLeft(32))
```

Comment:
Not necessarily related to this PR, but this seems like a great footgun for losing entropy. What happens if we just throw an exception in this case? 

We should at minimum file an issue to look at this again, or even better fix it on this PR.

Reply:
Fixed this in the most recent commit: https://github.com/bitcoin-s/bitcoin-s/pull/2936/commits/ac7b193d932fc87f9546d238cd0fbefd5e670934